### PR TITLE
Jvenstad/remove more deprecated deployment spec usage

### PR DESCRIPTION
**Pull request by jonmv**
@mpolden please review.

### CODE REVIEW

**Comment by jonmv** [comment]
@bratseth last commit.

**Comment by bratseth** [comment]
👍 re the last commit. So with this we support running multiple instances in prod?

**Comment by jonmv** [comment]
Yes.

**Comment by jonmv** [comment]
@bratseth feel free to review the now last commit: removing `singleInstance` things from `DeploymentSpec`.

File: controller-server/src/test/java/com/yahoo/vespa/hosted/controller/restapi/application/ApplicationApiTest.java [hunk 1]
```diff
@@ -661,17 +661,21 @@ public void testApplicationApi() {
...
-        // Sixth attempt has a multi-instance deployment spec, and fails.
+        // Sixth attempt has a multi-instance deployment spec, and is accepted.
         ApplicationPackage multiInstanceSpec = new ApplicationPackageBuilder()
                 .instances("instance1,instance2")
+                .systemTest()
+                .stagingTest()
                 .environment(Environment.prod)
                 .region("us-central-1")
                 .parallel("us-west-1", "us-east-3")
+                .endpoint("default", "foo", "us-central-1", "us-west-1", "us-east-3")
```
Comment (mpolden) [inline controller-server/src/test/java/com/yahoo/vespa/hosted/controller/restapi/application/ApplicationApiTest.java]: Is there a test that verifies endpoint configuration combined with multiple instances?

File: controller-server/src/main/java/com/yahoo/vespa/hosted/controller/deployment/DeploymentSteps.java [hunk 2]
```diff
@@ -39,9 +40,10 @@ public DeploymentSteps(DeploymentInstanceSpec spec, Supplier<SystemName> system)
 
     /** Returns jobs for this, in the order they are declared */
```
Comment (bratseth) [inline controller-server/src/main/java/com/yahoo/vespa/hosted/controller/deployment/DeploymentSteps.java]: Not: This javadoc should now say "in the order they should be executed" now.

**Review by bratseth — APPROVED** [review]
Great. Assuming you know it isn't too early to remove hasSingleInstance

File: controller-server/src/main/java/com/yahoo/vespa/hosted/controller/deployment/DeploymentSteps.java (hunk 2 above)
Comment (jonmv) [reply controller-server/src/main/java/com/yahoo/vespa/hosted/controller/deployment/DeploymentSteps.java]: Hehe, yes, I keep changing that doc back and forth :p Will replace the whole class next, to orchestrate instances, anyway.

**Comment by jonmv** [comment]
There are no usages of the deprecated stuff left, and I believe the only code that cares about test/staging being inside/outside of instances is the deployment orchestration code.

File: controller-server/src/test/java/com/yahoo/vespa/hosted/controller/restapi/application/ApplicationApiTest.java (hunk 1 above)
Comment (jonmv) [reply controller-server/src/test/java/com/yahoo/vespa/hosted/controller/restapi/application/ApplicationApiTest.java]: What calls are made against the rotation repository?

Comment (jonmv) [reply controller-server/src/test/java/com/yahoo/vespa/hosted/controller/restapi/application/ApplicationApiTest.java]: We haven't really supported multiple instances in the deployment spec before, so no. I'll add one. Can you think of other places such tests are needed?

Comment (jonmv) [reply controller-server/src/test/java/com/yahoo/vespa/hosted/controller/restapi/application/ApplicationApiTest.java]: Ah, endpoints.

**Comment by jonmv** [comment]
Adding a test now, but there is a hard-coded "ignore instance in global endpoint URLs" that's blocking me. Why is that there?

**Comment by jonmv** [comment]
@mpolden PTAL, note last to second commit.

**Comment by jonmv** [comment]
@mpolden PTAL :)

**Review by mpolden — APPROVED** [review]

**Pull request merged**